### ISSUE TITLE
Considering much faster alternatives

### ISSUE THREAD

**Issue by mikrosk** [atarijookie/ce-atari#65]
While I understand the point of #47, I'm not really excited to jump on board, from developer's point of view I'm fine with #64, no need to have "developer friendly" image for me. I'd like to discuss (and possibly look into) much faster alternatives because after quick research it's clear you will never be able to strip Raspbian & co under 15s because even with the kernel starting in 1-2s, the userspace will eat the rest, even with systemd. This assumes that every new feature (update) would be delivered on a separate image, yes.

Few links to look onto:
http://www.commodorepi.co.nr/ (boot + UI in a few seconds)
https://github.com/gamaral/rpi-buildroot (ditto)
http://raspberrypi.stackexchange.com/questions/14763/how-to-improve-boot-time-for-raspberry-pi (the proof why you can't break the ~10s boundary)

This is of course only a suggestion or better said an indication what I'd like to explore. I always try to look at stuff from user's perspective and as a user I want to have freaking fast boot time, ideally rendering #17 no longer needed.

**Comment by tin-nl** [comment]
That's the difficulty - fast boot time vs more support from developers. Cross compilation won't help with the latter, as people would like to do features via Python, using audio/video codecs etc. you name it (basicly with everything that's not included currently :) ). 
The current 6 seconds boot time "to ACSI" is nice, but needs some added seconds time until mount and Networking is available and also needs #17 to fully work.
I looked into the faster boots at the beginning (ramfs etc.), but the 6s wasn't that much slower, so it's unsure if it's worth the drawbacks.

Tuxie had some thoughts about a changeroot-based approach. So how about booting in a base system and start the full thingamajing from there? Would this be even possible?

**Comment by mikrosk** [comment]
Well, but how many people want that? As we discussed in the email -- developers could download a full featured Raspbian, install all the crazy stuff, develop, debug and test it and then _ask the maintainers to include it into the official image_. Which would have been delivered "over the air" as an update. I can't think of anything simpler, users would have the razor fast boot times and developers could play, too.

EDIT: I'd like to have something like a setup which boots into ACSI/Floppy-responding setup before the Atari device has the chance to complain, so basically something as #17 but much more straightforward...

**Comment by tin-nl** [comment]
To make it absolutely clear: I'd like that, too - my first suggestion was a buildroot/ramfs thingy. But we should acknowledge that esp. Python and PHP and whatnot tinkerers have a much less steep learning curve (and are probably used to) apt-get on their target and do their stuff. AFAIK we are not targeting professional developers here. They google, and 99% of the results (see your results in the other issue) are for Raspian.
OTOH you are absolutely right that giving people apt-get on an "embeded device" is a call for support hell. +1 for simplicity, too.

So, are you willing to give buildroot a go and do a test with that?

**Comment by mikrosk** [comment]
Maybe I shouldn't write in such rush, what I meant is that there would be two images to download (actually, as they are now):
1. User image, the super fast boot one with zero tweaking ability but starting as fast as the ST needs it to start.
2. Developer image, plain Raspbian really, slow as molasses but with the apt-get functionality. So you install python, php, mono ;), create your amazing addon and mail Jookie "hey man, so I made this solitaire clone, what about including this nice 200 MB set of dependencies on your super-fast distro?" and Jookie will be like "no pro, man" and released new image.

Of course, this assumes #47 update process, i.e. easy as hell and without any distro dependencies.

In short, yes, I'd like to take a look but it will take some time.

**Comment by atarijookie** [comment]
MiKRO 'promissed' again a much faster linux image in the private emails, which might or might not come in the future, so let's see if this will ever happen ;)

**Comment by mikrosk** [comment]
Shut up. ;) But this task summarises nicely what I'd like to do.

However, first I'd like to fix some annoying problems / missing features, regardless of the distro.

**Comment by atarijookie** [comment]
> However, first I'd like to fix some annoying problems / missing features, regardless of the distro.

I'm looking forward to this since I've gave you the device, mr. Collaborator ;)

**Comment by AdamKlob** [comment]
What about a dual-mode image when you can swith between developer mode/user mode on the fly?

**Comment by atarijookie** [comment]
@AdamKlob That would be awesome, thank you for offering yourself to try to do it :-D :+1:

**Comment by atarijookie** [comment]
Closing, when someone will work on this later, he can reopen it :)
